corehq/warehouse/transforms/sql/user_group_dim.sql

```diff
@@ -0,0 +1,43 @@
+INSERT INTO {{ user_group_dim }} (
+    domain,
+    user_dim_id,
+    group_dim_id,
+    dim_last_modified,
+    dim_created_on,
+    deleted
+)
+
+SELECT
+    ud.domain,
+    ud.id,
+    gd.id,
+    now(),
+    now(),
+    combined_user_group_table.deleted
+FROM
+(
+(
+    SELECT
+        UNNEST(user_ids) AS user_id,
+        group_id AS group_id,
+        false AS deleted
+    FROM
+        {{ group_staging }}
+)
```

Comment:
do we need to keep track of removed users for any reason on the user group dim? i think i'd be in favor of not keeping track of them since a deleted record here isn't really referenced anywhere on its own in the source.

if we don't keep track of them, then we just need to load the user group dim in two steps:

1) delete from user_group_dim using group_staging ... to delete any existing records for the staged groups
2) insert into user_group_dim select from group_staging join group_dim join user_dim ... to insert the current ones

and we should be able to get rid of the subquery